rows.go

```diff
@@ -126,6 +126,17 @@ func (r *Rows) AddRow(values ...driver.Value) *Rows {
 
 	row := make([]driver.Value, len(r.cols))
 	for i, v := range values {
+		// Convert user-friendly values (such as int or driver.Valuer)
+		// to database/sql native value (driver.Value such as int64)
+		var err error
+		v, err = driver.DefaultParameterConverter.ConvertValue(v)
```

Comment:
the interface for AddRow says it should be driver value arguments, if user is not certain whether the value is a valid db driver.Value, then it is his problem to test it. I do not like this **panic** statement. The library should not panic, if user mocks an invalid value for scanner then that is fine and on him, to be able to cope on these errors when reading the rows.

Reply:
user can use driver.Valuer when he is creating Rows to be mocked or abstract it so it would always evaluate these values. Or am I misunderstanding it?

Reply:
In reply to [the second comment](https://github.com/DATA-DOG/go-sqlmock/pull/129#discussion_r202686220):
> user can use driver.Valuer when he is creating Rows to be mocked or abstract it so it would always evaluate these values. Or am I misunderstanding it?

`driver.DefaultParameterConverter.ConvertValue(v)` includes support for `driver.Valuer`. By the way, if you have this kind of doubt, this highlight one hole in the testsuite. One test could be added using a `NullInt` as a value for `AddRow`.